src/eterna/mode/PoseEdit/PoseEditMode.ts

```diff
@@ -2357,7 +2357,20 @@ export default class PoseEditMode extends GameMode {
             this._helpBar.display.visible = true;
         };
 
-        if (hasNextPuzzle) {
+        if (this._puzzle.nextPuzzleLink != null) {
+            missionClearedPanel.nextButton.clicked.connect(() => {
+                keepPlaying();
+                if (Eterna.MOBILE_APP) {
+                    if (window.frameElement) {
+                        window.frameElement.dispatchEvent(
+                            new CustomEvent('navigate', {detail: this._puzzle.nextPuzzleLink})
```

Comment:
I didn't realize there were implications with this, but this won't work. The mobile app uses a separate navigation structure, so this will presumably either fail or navigate the user out of the app, both of which are a big problem... We should check with Sharif, but we may have to avoid allowing arbitrary links and only add a case for navigating to a collection page specifically